sky/templates/gcp-ray.yml.j2

```diff
@@ -15,6 +15,12 @@ docker:
     {%- if gpu is not none %}
       --gpus all
     {%- endif %}
+{%- if docker_login_config is not none %}
+  docker_login_config:
+    username: {{docker_login_config.username}}
+    password: {{docker_login_config.password}}
+    server: {{docker_login_config.server}}
+{%- endif %}
```

Comment:
Just want to make sure, we don't have gcp docker support w/ new provisioner until this PR..?

Reply:
yea... the private docker support does not work for GCP before this. We should probably add a smoke test for private docker image. Could you help file an issue and look into this?

Reply:
#3225